### PR TITLE
Add new boolean parameter stringify_facts

### PR DESCRIPTION
**Pull request by martinpfeifer**
This parameters must be set in the [main] or [agent] section not in the
[master] section. That's why specifing this in the server_additional_settings
parameter will not work.

### CODE REVIEW

File: templates/puppet.conf.erb [hunk 1]
```diff
@@ -64,3 +64,6 @@
 <% scope.lookupvar("puppet::additional_settings").sort_by {|k, v| k}.each do |key, value| -%>
     <%= key %> = <%= value %>
 <% end -%>
+<% if @stringify_facts == false -%>
```
Comment (ekohl) [inline templates/puppet.conf.erb]: Why only when it's false?

**Comment by ekohl** [comment]
We also have ```additional_settings``` for ```[main]``` and ```agent_additional_settings``` for ```[agent]``` precisely for this use case. I'm fine with this change though, especially since it fixes an incorrect example.

File: templates/puppet.conf.erb (hunk 1 above)
Comment (martinpfeifer) [inline templates/puppet.conf.erb]: The puppet internal default is currently true (see https://docs.puppetlabs.com/references/latest/configuration.html#stringifyfacts). I thought if the parameter doesn't deviate from the default (set to true or undef) i don't set it in the config at all.

**Comment by martinpfeifer** [comment]
I added this a while ago locally but didn't push to GitHub. In the meantime https://github.com/FILIADATAGmbH/puppet-puppet/commit/4cc6629697c111b73aee628f41158f0e7db81bbb added what caused my changes initially.
Not sure if its better to have this as explicit parameter or handle it via the generic additional settings interface.

**Comment by ekohl** [comment]
I think that mostly depends on how common it is to set the parameter and I can't really judge about this one.

File: templates/puppet.conf.erb (hunk 1 above)
Comment (mmoll) [inline templates/puppet.conf.erb]: I think this is OK, as the default is ```true``` now and the whole option will be gone in Puppet 4 by puppetlabs/puppet#3212

**Comment by mmoll** [comment]
A test would be nice :slot_machine:

**Comment by martinpfeifer** [comment]
I now think the generic interface is sufficient to configure this setting as it's probably not that common to configure it at all. I'm closing this pull request and create a new one that fixes only the documentation.

**Pull request closed without merging**